tests/test_topology.py

```diff
@@ -41,42 +41,276 @@ def test_parse(self):
             side_effect = hwloc_calc_values,
         ):
             st = SystemTopology(load = False)
-            st._parse(filename = 'tests/data/intel-core-i7-4790.xml')
+            st._parse(filename = 'tests/data/single-intel-core-i7-4790.xml')
 
             subprocess.check_output.assert_has_calls([
                 call(args=[
                     'hwloc-calc', '-I', 'Package', '--physical-input', '--logical-output',
                     'Package:0'
                 ]),
                 call(args=[
-                    'hwloc-calc', '-I', 'Core',    '--physical-input', '--logical-output',
+                    'hwloc-calc', '-I', 'Core', '--physical-input', '--logical-output',
                     'Package:0.Core:0',
                     'Package:0.Core:1',
                     'Package:0.Core:2',
                     'Package:0.Core:3'
                 ]),
                 call(args=[
-                    'hwloc-calc', '-I', 'PU',      '--physical-input', '--logical-output',
+                    'hwloc-calc', '-I', 'PU', '--physical-input', '--logical-output',
                     'Package:0.Core:0.PU:0', 'Package:0.Core:0.PU:4',
                     'Package:0.Core:1.PU:1', 'Package:0.Core:1.PU:5',
                     'Package:0.Core:2.PU:2', 'Package:0.Core:2.PU:6',
                     'Package:0.Core:3.PU:3', 'Package:0.Core:3.PU:7'
                 ])
             ])
 
-            # There is 1 package.
+            # The machine has 1 package.
             assert len(st.packages) == 1
+            assert st.get_num_packages() == 1
 
             # The package has 4 cores.
             assert len(st.packages[0].cores) == 4
+            assert st.packages[0].get_num_cores() == 4
 
-            # Each core has 2 PUs.
+            # The first core of the package has 2 PUs.
             assert len(st.packages[0].cores[0].pus) == 2
+            assert st.packages[0].cores[0].get_num_pus() == 2
 
             # The first PU of the second core of the package has OS index 1 and logical index 2.
             assert st.packages[0].cores[1].pus[0].os_index      == 1
             assert st.packages[0].cores[1].pus[0].logical_index == 2
 
+            gen_cores = st.recurse_cores()
+            _ = next(gen_cores)
+            second_core = next(gen_cores)
+            assert second_core.pus[0].logical_index == 2
```

Comment:
I think this essentially the test that's done a few lines above. Here, I'm essentially doing the same test, but using now the generators coming from `recurse_cores` and `recurse_pus`. The idea is to test these generators. I've added this to the docstring.

Reply:
Ok, but instead of calling `next` yourself repeatedly, let's do:
```python
import itertools

class Tester:
    def test(self):
        for pu in ['pu-0', 'pu-1', 'pu-2']:
            yield pu

t = Tester()
print(next(itertools.islice(t.test(), 2, None)))
```

Reply:
OK! That's a nice suggestion!